viewing_party/party.py

```diff
@@ -1,23 +1,226 @@
+    if True: 
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+    if title == None or genre == None or rating == None:
+        return None
+    return movie_dict
+
+def add_to_watched(user_data, movie):
+    '''
+    Input: user_data is a dictionary with a key "watched" and value of list of movie dictionaries the user has watched
+    Output: Function adds movie dictionary to user_data and returns user_data
+    '''
+    user_data["watched"] = [movie]
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    '''
+    Input: user_data is list of movie dictionaries with key "watchlist" and value of list of movie dictionaries
+    that the user wants to watch
+    Output: Function adds movie dictionary to "watchlist" and returns user_data
+    '''
+    user_data["watchlist"] = [movie]
+    return user_data
+
+def watch_movie(user_data, title):
+    '''
+    Input: user dictionary is a list of movie dictionaries with "watched" and "watchlist keys
+    representing the movies the user has watched or wants to watch. Title is a string representing
+    a movie the user has watched. 
+    Output: function moves movie represented by title to watched list if the user has that movie in the watchlist.
+    Function returns user_data.
+    '''
+    for movie_index in range(len(user_data["watchlist"])):
+        for title_key,current_title in user_data["watchlist"][movie_index].items():
+            if current_title == title:
+                user_data["watched"].append(user_data["watchlist"][movie_index])
+                user_data["watchlist"].pop(movie_index)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    '''
+    Input: user_data is a dictionary with key "watched" and values of a list of movie dictionaries
+    (containing title, gener, and rating), representing movies the user has watched. 
+    Output: function finds the average rating of all movies in watched list and returns that average rating. 
+    '''
+    sum = 0
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+    try:
+        average = sum / len(user_data["watched"])
+    except ZeroDivisionError:
+        average = 0.0
+    return average
+
+def get_most_watched_genre(user_data):
+    '''
+    Input: user_data is a dictionary with a "watched" list of movie dictionaries. Each movie has a key "genre"
+    Output: Function determines which genre is most frequent and returns that genre value. If "watched"
+    is an empty list, function returns None.
+    '''
+    genre_list = []
+    genre_count = {}
+
+    if user_data["watched"] == []:
+        return None
+
+    for movie in range(len(user_data["watched"])):
+        genre_list.append(user_data["watched"][movie]['genre'])
+        
+    for i in genre_list:

```

Comment:
This loop iterates over the same data as the first loop. Now that we have talked about Big O, we know that it would be more efficient to only loop over this genre data once.

I would recommend compiling the list of genres and their counts all in the same loop.